lib/network/rest/RESTClient.js

```diff
@@ -210,6 +210,44 @@ class RESTClient {
     }
 
     /**
+     * send a GET request to the REST server
+     * @param {String} action - The action to query
+     * @param {String} reqUids - The serialized request ids
+     * @param {RESTClient~getCallback} callback - callback
+     * @returns {undefined}
+     */
+    getAction(action, reqUids, callback) {
+        const log = this.createLogger(reqUids);
+        const headers = {};
+        setRequestUids(headers, reqUids);
+        const reqParams = {
+            hostname: this.host,
+            port: this.port,
+            method: 'GET',
+            path: `${constants.dataFileURL}?${action}`,
+            headers,
+            agent: this.httpAgent,
+        };
+        log.debug('about to send GET request', {
+            hostname: reqParams.hostname,
+            port: reqParams.port,
+            path: reqParams.path,
+            headers: reqParams.headers });
+
+        const request = http.request(reqParams, response => {
+            response.once('readable', () => {
+                if (response.statusCode !== 200 &&
+                    response.statusCode !== 206) {
+                    return callback(makeErrorFromHTTPResponse(response));
+                }
+                return callback(null, response.read().toString());
```

Comment:
Note that this should only work for responses 16kb or less (the highwatermark for the internal buffer of the stream). You might want to document that limit for the function?